src/react-components/src/components/01-atoms/Links/LinkHelper.tsx

```diff
@@ -0,0 +1,57 @@
+// Base Link Helper Class
+import * as React from "react";
+
+export interface LinkHelperProps {
+  className: string;
+  icon?: { element: JSX.Element };
+  url?: string;
+  attributes?: {};
+}
+
+export default class LinkHelper extends React.Component<LinkHelperProps, {}> {
+  constructor(props: LinkHelperProps) {
+    super(props);
+  }
+
+  render(): JSX.Element {
+    const { url, className, attributes, icon } = this.props;
+
+    let props = {
+      className: className,
+      ...attributes
+    };
+
+    if (React.Children.count(this.props.children) > 1) {
+      // There should only be one child
+      // but React.Children.only() can't cast to React.ReactElement for typechecking
+      // Therefore, React.Children.map() must be used.
+
+      throw new Error(`Please only pass one child, got ${this.props.children}`);
+    }
```

Comment:
You can cast `this.props.children` as a `React.ReactElement` for `React.Children.only()`. I tried that thinking it would make things easier but then ran into issues when the child was a _string_. So I played around with it for a while and found you can try something like:

```
    let child: React.ReactElement;
    // If the value is a string, then it's only one element but wrap it in a fragment.
    if (typeof this.props.children === 'string') {
      child = <>{this.props.children}</>;
    } else {
      // otherwise it is a react element but could have multiple children. If that's
      // the case, you can check that it's only one element.
      child = this.props.children as React.ReactElement;
      try {
        React.Children.only(child as React.ReactElement)
      } catch(e) {
        // Catching the error because React's error isn't as helpful.
        // This will tell you _where_ the error is happening.
        throw new Error(`LinkHelper: ${e}`);
      }
    }
```

If you do this, then you can use `child` like you are doing below without having to do the `.map` since you know it's only one element. And you can return `link` (not `link[0]`) if you assign it `React.createElement` or `React.cloneElement`.

Is this cleaner or easier to understand than your solution? I don't know but this is just an alternative (and I also wanted to find out how `React.Children.only` works). Your solution is fine as is, especially if it's easier for you to understand.

Reply:
oh my gosh this is so beautiful thank you
My brain's been on a hamster wheel about this all morning